testbed_api/camera.py

```diff
@@ -7,30 +10,58 @@ class VideoCamera:
 
     def __init__(self, index: int = 0):
         self._capture = cv2.VideoCapture(index)
+        self._capture.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc('X', 'V', 'I', 'D'))
+        self._prev_frame_time = 0
+        self._new_frame_time = 0
+        
+        self._frames = deque()
+        self._thread = threading.Thread(target = self.read_frames)
+        self._thread.start()
```

Comment:
Reading from the camera takes place in a separate thread, all results are written to a deque `self_frames`.